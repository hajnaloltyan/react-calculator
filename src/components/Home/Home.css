.page-title {
  margin-bottom: 40px;
  font-size: 30px;
}
