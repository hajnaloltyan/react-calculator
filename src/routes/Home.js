import React from 'react';

const Home = () => (
  <section>
    <h1>Home</h1>
  </section>
);

export default Home;
