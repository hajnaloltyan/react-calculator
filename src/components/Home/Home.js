import React from 'react';
import './Home.css';

const Home = () => (
  <section>
    <h2>
      Welcome to Our Page!
    </h2>

    <p>
      At Math Magicians, we&apos;re passionate about making math accessible and exciting
      for everyone. With our state-of-the-art React calculator, you can effortlessly
      solve equations and explore mathematical concepts. Whether you&apos;re a student,
      professional, or simply fascinated by numbers, our calculator is here to assist
      you every step of the way.
    </p>

    <p>
      But Math Magicians is more than just a calculator. We provide a platform for you to
      expand your mathematical knowledge, discover fascinating facts, and connect with a
      community of like-minded individuals. From insightful quotes about computers to
      interactive features, our user-friendly interface is designed to make your mathematical
      journey enjoyable and rewarding.
    </p>

    <p>
      Join us on this exciting adventure, where every problem has a solution, and every solution
      holds a world of possibilities. Welcome to Math Magicians - where math comes to life!
    </p>
  </section>
);

export default Home;
