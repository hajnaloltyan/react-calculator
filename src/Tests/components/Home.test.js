import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import Home from '../../components/Home/Home';

describe('Test the Home component', () => {
  test('Check if the Home component render correctly', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Check that the home has te correct title', () => {
    const homeComponent = render(<Home />);
    expect(homeComponent.container).toHaveTextContent('Welcome to Our Page!');
  });

  test('Check that the home has te correct last paragraph', () => {
    const homeComponent = render(<Home />);
    expect(homeComponent.container).toHaveTextContent('Join us on this exciting adventure, where every problem has a solution, and every solution holds a world of possibilities. Welcome to Math Magicians - where math comes to life!');
  });
});
