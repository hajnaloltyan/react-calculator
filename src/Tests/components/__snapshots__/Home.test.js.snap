// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Home component Check if the Home component render correctly 1`] = `
<section>
  <h2
    className="page-title"
  >
    Welcome to Our Page!
  </h2>
  <p>
    At Math Magicians, we're passionate about making math accessible and exciting for everyone. With our state-of-the-art React calculator, you can effortlessly solve equations and explore mathematical concepts. Whether you're a student, professional, or simply fascinated by numbers, our calculator is here to assist you every step of the way.
  </p>
  <p>
    But Math Magicians is more than just a calculator. We provide a platform for you to expand your mathematical knowledge, discover fascinating facts, and connect with a community of like-minded individuals. From insightful quotes about computers to interactive features, our user-friendly interface is designed to make your mathematical journey enjoyable and rewarding.
  </p>
  <p>
    Join us on this exciting adventure, where every problem has a solution, and every solution holds a world of possibilities. Welcome to Math Magicians - where math comes to life!
  </p>
</section>
`;
